src/robot/model/testcase.py

```diff
@@ -34,8 +39,11 @@ class TestCase(ModelObject):
     repr_args = ('name',)
     __slots__ = ['parent', 'name', 'doc', 'timeout', 'lineno', '_setup', '_teardown']
 
-    def __init__(self, name='', doc='', tags=None, timeout=None, lineno=None,
-                 parent=None):
+    def __init__(self, name: str = '', doc: str = '',
+                 tags: Union[List[str], str, None] = None,
```

Comment:
`tags` can also be `Tags` instance. It could also be a tuple or any other iterable but I'm not sure is that relevant.